### ISSUE TITLE
Add MyPy

### ISSUE THREAD

**Issue by Peter9192** [AI4S2S/s2spy#14]
In #12 I started adding type annotations. However, mypy hasn't yet been set up for this repo, so they haven't been checked yet. It would be good to configure mypy asap, so as to avoid accumulating many type issues

**Comment by BSchilperoort** [comment]
I quickly installed mypy to test the package. Things mostly go OK, however some of our dependencies do not having type hinting.

```
s2spy\traintest.py:8: error: Skipping analyzing "pandas": module is installed, but missing library stubs or py.typed marker
s2spy\traintest.py:10: error: Skipping analyzing "sklearn.model_selection._split": module is installed, but missing library stubs or py.typed marker
```

**Comment by BSchilperoort** [comment]
Currently I get no mypy errors when running mypy locally.
I have the exact same versions of Python (3.10.5) mypy (0.971) and mypy-extensions installed. Local machine is windows.

<hr>

One of the two errors that GH actions shows is the following;
```
s2spy/_resample.py
  Line: 140
    mypy: error / Argument 1 to "rename" of "Dataset" has incompatible type "str"; 
                  expected "Optional[Mapping[Any, Hashable]]" (col 48)
```

Which is strange because this is within an if isinstance statement;
```python
139    if isinstance(input_data, xr.DataArray) and interval_means.name is None:
140        interval_means = interval_means.rename("mean_values")
```

**Comment by Peter9192** [comment]
Hmm that's curious. [This article](https://adamj.eu/tech/2021/05/17/python-type-hints-how-to-narrow-types-with-isinstance-assert-literal/) suggests that mypy should be able to infer the narrowed type.

**Comment by BSchilperoort** [comment]
I now know why this error shows up. I test for the type of input_data and not of interval_means.

They will have the same type, but mypy is not able to infer this, and the if statement should probably check for the type of interval_means and not input_data.

However, mypy is still not giving me any errors locally. When using `reveal_type()` in the code, mypy does however print this to the terminal:
```
>>> mypy . --ignore-missing-imports --exclude=build

s2spy\_resample.py:141: note: Revealed type is "xarray.core.dataarray.DataArray"
Success: no issues found in 16 source files
```

**Comment by Peter9192** [comment]
Merged as part of #60
